batch_requests.go

```diff
@@ -40,7 +41,18 @@ type BatchRequestHandler struct {
 
 // doRequest will make the same request but return a BatchReplyUnit
 func (b *BatchRequestHandler) doRequest(req *http.Request, relURL string) BatchReplyUnit {
-	resp, err := http.DefaultClient.Do(req)
+	tr := &http.Transport{TLSClientConfig: &tls.Config{}}
+	if cert := getUpstreamCertificate(req.Host, b.API); cert != nil {
+		tr.TLSClientConfig.Certificates = []tls.Certificate{*cert}
+	}
+
+	if config.Global.ProxySSLInsecureSkipVerify {
```

Comment:
Can be replaced by `tr.TLSClientConfig.InsecureSkipVerify = config.Global.ProxySSLInsecureSkipVerify`